### PR TITLE
Fixed #27333 -- Prevented BASE64 encoding in message.as_string() on Python 3

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: tests/mail/tests.py [hunk 1]
```diff
@@ -574,6 +574,9 @@ def test_dont_base64_encode(self):
         s = msg.message().as_bytes()
         self.assertNotIn(b'Content-Transfer-Encoding: quoted-printable', s)
         self.assertIn(b'Content-Transfer-Encoding: 8bit', s)
+        s = msg.message().as_string()
+        self.assertNotIn(str('Content-Transfer-Encoding: quoted-printable'), s)
```
Comment (timgraham) [inline tests/mail/tests.py]: Is it possible that the header can be duplicated? If not, I don't see the value in the NotIn assertions (perhaps you can explain).

Comment (claudep) [inline tests/mail/tests.py]: I blindly copied the existing tests, so yes, we could probably remove the `assertNotIn` lines.

**Comment by claudep** [comment]
I think this is already a welcome simplification. We might do more after dropping Python 2, but this is true for other parts of Django and I don't think it brings much value opening tickets for that now.

**Comment by timgraham** [comment]
Okay, looks good. I thought the `assertNotIn` should be removed in a separate commit, there are a few more to clean up, anyway.

**Comment by claudep** [comment]
Not sure, do you want me to readd the `assertNotIn` in this patch?

**Comment by timgraham** [comment]
How about pushing this patch first? http://dpaste.com/2MX3RK1

**Comment by claudep** [comment]
Why not. However, semantically, we are testing that the encoding is not quoted-printable, it can be anything else. So +0.

**Comment by timgraham** [comment]
Right, I think the comments saying "Shouldn't use quoted printable" provides sufficient clarification about that.

**Pull request merged**